src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -378,6 +376,28 @@ public BlockRenderLayer getBlockLayer()
         return BlockRenderLayer.SOLID;
     }
 
+    /**
+     * Registery block at gameregistry.
+     * @param registry the registry to use.
+     * @return the block itself.
+     */
+    public BlockMinecoloniesRack registerBlock(final IForgeRegistry<Block> registry)
+    {
+        registry.register(this);
+        return this;
+    }
+
+    /**
+     * Registery block at gameregistry.
+     * @param registry the registry to use.
+     * @return the block itself.
+     */
+    public Block registerItemBlock(final IForgeRegistry<Item> registry)
+    {
+        registry.register((new ItemBlock(this)).setRegistryName(this.getRegistryName()));
+        return this;
+    }
+
     public enum EnumType implements IStringSerializable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)